packages/babel-plugin-proposal-class-static-block/src/index.js

```diff
@@ -25,10 +30,19 @@ export default declare(({ types: t, template, assertVersion }) => {
   return {
     name: "proposal-class-static-block",
     inherits: syntaxClassStaticBlock,
+
+    pre() {
+      // Enable this in @babel/helper-create-class-features-plugin, so that it
+      // can be handled by the private fields and methods transform.
+      enableFeature(this.file, FEATURES.staticBlocks, /* loose */ false);
```

Comment:
Would static blocks be transformed by class-feature plugin if only privateIn is transformed? If so we can enable this feature only when `class-properties` is required from given targets.

Reply:
> Would static blocks be transformed by class-feature plugin if only privateIn is transformed?

No, because the `private-in` plugin doesn't enable the class features helper (it only enables the _feature_ if the helper is already enabled).

> If so we can enable this feature only when `class-properties` is required from given targets.

This would still fail in the case described by the original PR description.